erts/emulator/beam/erl_bif_lists.c

```diff
@@ -235,6 +235,11 @@ static Eterm append(Export *bif_entry, BIF_ALIST_2) {
          * in 'not_a_list'. */
         return rhs;
     } else if (is_list(lhs)) {
+        /* Skip appending for `Lhs ++ []`: return `Lhs` instead. */
```

Comment:
Should this be applied to `List -- []` as well?

Reply:
Yes.